src/main/java/Duke.java

```diff
@@ -1,10 +1,246 @@
+import duke.command.AException;
+import duke.command.Parser;
+import duke.command.Storage;
+import duke.command.TaskList;
+import duke.command.Ui;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import java.util.Scanner;
+import java.io.FileNotFoundException;
+import java.util.LinkedList;
+
+/**
+ * Main driver class for Duke.
+ */
 public class Duke {
+    
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+
+    /**
+     * Creates a Duke object that takes in a filepath to
+     * the file where the list of previous tasks is stored.
+     * @param filePath path to where the last saved task list is stored
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.printFileContents());
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+        }
+
+    }
+
+    //write to file when there is a change in state
+
+    /**
+     * Runs the logic for Duke.
+     */
+    public void run() {
+        Scanner scan = new Scanner(System.in);
+        
+        ui.showWelcome();
+
+        //LinkedList used to store all the String given by the user;
+        // LinkedList<Task> li = new LinkedList<Task>();
+
+        //read the existing task from the file and create the Duke.tasks to put
+        //into the list. 
+
+        // try{
+        //     storage.printFileContents(li);
+        // } catch(FileNotFoundException e){
+        //     //System.out.println("File not found");
+        // }
+
+        while (true) {
+            ui.printnewline();
+            String echo = scan.nextLine();
+            Parser split = new Parser(echo);
+            String error = "";
+
+            String firstWord = split.getType();
+
+            if (firstWord.equals("bye")) {
+                // if bye, then end the program
+                ui.printline();
+                System.out.println("\tBye. Hope to see you again soon!");
```

Comment:
Since you already have a `UI` class, you may want to use that class to handle all interactions with user (i.e. handle all input and output). It might be a bit confusing if sometimes you print from `UI` class and sometimes directly from `Duke`.